openlcs/libs/corgi.py

```diff
@@ -238,3 +259,93 @@ def get_paginated_data(self, query_params=None, api_path="components"):
                 # general exceptions, timeout/connection/maxredirect etc.
                 logger.error("Request failed: %s", e)
                 break
+
+    def get_srpm_component(self, component):
+        """
+        Returns the source rpm component for `component`
+        """
+        if component["arch"] == "src":
+            return component
+        else:
+            sources = component.get("sources")
+            if not sources:
```

Comment:
I checked the container that you mentioned in the script, use corgi-stage, but I found each of them have no `sources` item